runbotics/runbotics-orchestrator-ui/src/src-app/views/credentials/Credential/EditCredential/CredentialAttributeList/CredentialAttributeList.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC, useEffect } from 'react';
+
+import { CircularProgress, Grid } from '@mui/material';
+
+import { useDispatch, useSelector } from '#src-app/store';
+
+import { credentialsActions } from '#src-app/store/slices/Credentials';
+import { fetchAllTemplates } from '#src-app/store/slices/CredentialTemplates/CredentialTemplates.thunks';
+
+import { BasicCredentialDto } from '../../Credential.types';
+import { DisplayAttribute } from '../CredentialAttribute/Attribute.types';
+
+import TemplateAttribute from '../CredentialAttribute/TemplateAttribute/TemplateAttribute';
+
+
+interface CredentialAttributesListProps {
+    credential: BasicCredentialDto;
+    templateId: string;
+    isNewCredential: boolean;
+}
+
+const CredentialAttributesList: FC<CredentialAttributesListProps> = ({ credential, templateId, isNewCredential }) => {
+    const dispatch = useDispatch();
+    const templates = useSelector(state => state.credentialTemplates.data);
+    const credentialTemplate = templates.find(template => template.id === templateId);
+    const templatesLoading = useSelector(state => state.credentialTemplates.loading);
+
+    useEffect(() => {
+        dispatch(credentialsActions.fetchOneCredential({resourceId: credential.id}));
+    }, []);
+
+    useEffect(() => {
+        if (templates.length === 0) {
+            dispatch(fetchAllTemplates());
+        }
+    }, [dispatch, templates]);
+
+    if (templatesLoading || templates.length === 0) {
```

Comment:
line 38: remove templates.length from condition

Reply:
As discussed - needed for credentialTemplate variable